src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -105,23 +110,33 @@ case class CreateDeltaTableCommand(
 
         val data = Dataset.ofRows(sparkSession, query.get)
 
-        val options = new DeltaOptions(table.storage.properties, sparkSession.sessionState.conf)
+        val options = operation match {
+          case TableCreationModes.ReplaceTable | TableCreationModes.CreateOrReplaceTable =>
+            val options = table.storage.properties ++ Map("overwriteSchema" -> "true")
+            if (txn.readVersion > -1L) {
+              // Otherwise the metadata will be updated in WriteIntoDelta
```

Comment:
What's this an "otherwise" to?

Reply:
`txn.readVersion == -1L`, i.e. empty Delta table. I'll add a comment